extractor/filesystem/language/python/uvlock/uvlock.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package uvlock extracts uv.lock files.
+package uvlock
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/python/internal/pypipurl"
+	"github.com/google/osv-scalibr/extractor/filesystem/osv"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+type uvLockPackageSource struct {
+	Virtual  string `toml:"virtual"`
+	Git      string `toml:"git"`
+	Registry string `toml:"registry"`
```

Comment:
This one doesn't seem used so I'd not add it to the type here (unless we want to write it to e.g. the metadata)

Reply:
I don't think we need this at the moment in the metadata, we are assuming all these packages are from PyPI, otherwise the Ecosystem() won't be correct. 

We could add a string check to filter out non PyPI packages, but that's probably too strict and cause breakages if the url changes slightly.

Let's just remove this field imo.